### PR TITLE
refresh token for external endpoints

### PR DESCRIPTION
**Pull request by brianyingling**
Fix to allow endpoints in config to accept refresh tokens
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23issuecomment-69276293%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23issuecomment-69276405%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23issuecomment-69276954%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23discussion_r22695830%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23issuecomment-69282375%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23issuecomment-69276293%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20you%20submit%20CLA%20for%20msopentech%20for%20us%20to%20take%20this%20pull%20request%3F%20You%20only%20need%20to%20do%20this%20once.%22%2C%20%22created_at%22%3A%20%222015-01-09T00%3A39%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20thought%20I%20did.%20I%27ll%20resubmit.%22%2C%20%22created_at%22%3A%20%222015-01-09T00%3A40%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3287568%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brianyingling%22%7D%7D%2C%20%7B%22body%22%3A%20%22Submitted.%22%2C%20%22created_at%22%3A%20%222015-01-09T00%3A46%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3287568%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brianyingling%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%21%22%2C%20%22created_at%22%3A%20%222015-01-09T01%3A51%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3287568%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brianyingling%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ae0e35b21b24082312c52608cd0fae5438c73e5d%20lib/adal-angular.js%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/66%23discussion_r22695830%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20you%20put%20the%20for%20loop%20inside%20%5Cr%5Cnif%20%28authService.config%29%7B%5Cr%5Cn%20///inside%20here%5Cr%5Cn%7D%22%2C%20%22created_at%22%3A%20%222015-01-09T00%3A57%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/adal-angular.js%3AL248-271%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/66#issuecomment-69276293'>General Comment</a></b>
- <a href='https://github.com/omercs'></a> Can you submit CLA for msopentech for us to take this pull request? You only need to do this once.
- <a href='https://github.com/brianyingling'></a> I thought I did. I'll resubmit.
- <a href='https://github.com/brianyingling'></a> Submitted.
- <a href='https://github.com/brianyingling'></a> done!
- [x] <a href='#crh-comment-Pull ae0e35b21b24082312c52608cd0fae5438c73e5d lib/adal-angular.js 11'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/66#discussion_r22695830'>File: lib/adal-angular.js:L248-271</a></b>
- <a href='https://github.com/omercs'></a> Can you put the for loop inside
if (authService.config){
///inside here
}

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/66?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/66?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/66'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/66'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by omercs** [comment]
Can you submit CLA for msopentech for us to take this pull request? You only need to do this once.

**Comment by brianyingling** [comment]
I thought I did. I'll resubmit.

**Comment by brianyingling** [comment]
Submitted.

File: lib/adal-angular.js [hunk 1]
```diff
@@ -248,15 +248,23 @@ if (typeof module !== 'undefined' && module.exports) {
...
                         var resource = authService.getResourceForEndpoint(config.url);
                         var tokenStored = authService.getCachedToken(resource);
+                        var isEndpoint = false;
                         if (tokenStored) {
                             // check endpoint mapping if provided
                             config.headers.Authorization = 'Bearer ' + tokenStored;
                             return config;
                         } else {
+
+                            for (var endpointUrl in authService.config.endpoints) {
```
Comment (omercs) [inline lib/adal-angular.js]: Can you put the for loop inside 
if (authService.config){
 ///inside here
}

**Comment by brianyingling** [comment]
done!

**Pull request merged**